### PR TITLE
fix(cli): close test runner once all tests complete

### PR DESCRIPTION
**Pull request by caspervonb**
Closes https://github.com/denoland/deno/issues/9603

### CODE REVIEW

File: cli/tools/test_runner.rs [hunk 1]
```diff
@@ -89,6 +89,14 @@ pub fn render_test_file(
     options
   ));
 
+  test_file.push_str(
+    r"
+if ('close' in globalThis) {
+  globalThis.close();
+}
+    ",
+  );
```
Comment (bartlomieju) [inline cli/tools/test_runner.rs:98]: I don't think this is correct; tests are not run in worker context

**Comment by caspervonb** [comment]
Blocked on https://github.com/denoland/deno/issues/8713

File: cli/tools/test_runner.rs (hunk 1 above)
Comment (caspervonb) [reply cli/tools/test_runner.rs]: Wasn't sure if unit tests invoked the manifest or not (seemingly not).

**Comment by caspervonb** [comment]
Incompatible with #9815, needs to be rewritten.
Closing without merge.

**Pull request closed without merging**